### PR TITLE
16.0 fix resize chart mamu

### PR DESCRIPTION
**Pull request by mamu-odoo**
## Description:

Add a maximum to deltaX and deltaY to prevent the figure from moving.

Odoo task ID : [3044939](https://www.odoo.com/web#cids=1&menu_id=5879&action=333&active_id=2328&model=project.task&view_type=form&id=3044939)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/figures/figure/figure.ts [hunk 1]
```diff
@@ -269,8 +269,8 @@ export class FigureComponent extends Component<Props, SpreadsheetChildEnv> {
     this.dnd.height = figure.height;
 
     const onMouseMove = (ev: MouseEvent) => {
-      const deltaX = dirX * (ev.clientX - initialX);
-      const deltaY = dirY * (ev.clientY - initialY);
+      const deltaX = Math.max(dirX * (ev.clientX - initialX), MIN_FIG_SIZE - figure.width);
+      const deltaY = Math.max(dirY * (ev.clientY - initialY), MIN_FIG_SIZE - figure.height);
```
Comment (LucasLefevre) [inline src/components/figures/figure/figure.ts:273]: Then the two `Math.max` below are useless and can be removed

File: tests/components/figure.test.ts [hunk 2]
```diff
@@ -38,6 +39,21 @@ function createFigure(
   });
 }
 
+const anchorSelectors = {
+  top: ".o-anchor.o-top",
```
Comment (pro-odoo) [inline tests/components/figure.test.ts:43]: top, right, bottom and left are not necessary. If we add them, we will have to maintain it for no reason.

File: src/components/figures/figure/figure.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ type Anchor =
 const ANCHOR_SIZE = 8;
 const BORDER_WIDTH = 1;
 const ACTIVE_BORDER_WIDTH = 2;
-const MIN_FIG_SIZE = 80;
+export const MIN_FIG_SIZE = 80;
```
Comment (pro-odoo) [inline src/components/figures/figure/figure.ts:29]: As we have to export it, I would move it to `constants.ts`

**Review by pro-odoo — CHANGES_REQUESTED** [review]
Ok except the two little comments :)

File: tests/components/figure.test.ts (hunk 2 above)
Comment (pro-odoo) [reply tests/components/figure.test.ts:42]: This comment is marked as resolved but without changes nor explanation. Is it intended ?

**Review by pro-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**